src/rules/noRequireImportsRule.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "require() style import is forbidden";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new NoRequireImportsWalker(sourceFile, this.getOptions()));
+    }
+}
+
+class NoRequireImportsWalker extends Lint.RuleWalker {
+    public visitVariableStatement(node: ts.VariableStatement) {
+        const declarations = node.declarationList.declarations;
+        for (let decl of declarations) {
+            this.processDeclaration(decl);
+        }
+        super.visitVariableStatement(node);
+    }
+
+    public visitImportEqualsDeclaration(node: ts.ImportEqualsDeclaration) {
+        this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.FAILURE_STRING));
+        super.visitImportEqualsDeclaration(node);
+    }
+
+    private processDeclaration(decl: ts.VariableDeclaration) : void {
```

Comment:
in a few other places, we use the verb "handle", so let's make this `handleDeclaration`. also omit the return type